backend/app/mailers/enrollment_mailer.rb

```diff
@@ -45,6 +45,20 @@ def notify_support_franceconnect
     )
   end
 
+  def notification_email_unknown_software
+    @target_api_label = data_provider_config["label"]
+    @enrollment = Enrollment.find(params[:enrollment_id])
```

Comment:
`params[:enrollment_id]` and `params[:enrollment_id]` are not defined here. You should add this params when calling this fonction with `EnrollmentMailer.with(...)`